### PR TITLE
Adds order/orderItem dao methods

### PR DESCRIPTION
**Pull request by madisoncarr**

### CODE REVIEW

File: src/main/java/com/games/online/shop/service/dao/GameShopDao.java [hunk 1]
```diff
@@ -46,4 +64,43 @@ public void deleteMechanics(Mechanics mechanics) {
...
+            log.error("Error removing order item for order={}", orderId, ex);
+        }
+    }
+
+    private void setOrderItemId(Long orderId, OrderItem orderItem) {
+        try {
+            if (orderItem.getGame() != null) {
+                OrderItemId orderItemId = new OrderItemId(orderId, orderItem.getGame().getId());
+                orderItem.setId(orderItemId);
+            }
```
Comment (mbarzilai) [inline src/main/java/com/games/online/shop/service/dao/GameShopDao.java:107]: Should we throw an exception if the game is null?

File: src/main/java/com/games/online/shop/service/dao/GameShopDao.java [hunk 2]
```diff
@@ -46,4 +64,43 @@ public void deleteMechanics(Mechanics mechanics) {
...
+    public void addOrderItem(Long orderId, OrderItem orderItem) {
+        try {
+            Optional<Order> order = orderRepository.findById(orderId);
+            if (order.isPresent()) {
+                setOrderItemId(orderId, orderItem);
+                orderItem.setOrder(order.get());
+                order.get().addOrderItem(orderItem);
+                orderRepository.save(order.get());
+                orderItemRepository.save(orderItem);
+            }
```
Comment (mbarzilai) [inline src/main/java/com/games/online/shop/service/dao/GameShopDao.java:79]: What should happen if the order isn't present? The method calling this item shouldn't think that it was successful, right?

File: src/main/java/com/games/online/shop/service/dao/GameShopDao.java [hunk 3]
```diff
@@ -46,4 +64,43 @@ public void deleteMechanics(Mechanics mechanics) {
...
+    }
+
+    public void deleteOrderItem(Long orderId, OrderItem orderItem) {
+        try {
+            Optional<Order> order = orderRepository.findById(orderId);
+            if (order.isPresent()) {
+                order.get().getOrderItems().remove(orderItem);
+                orderRepository.save(order.get());
+                orderItemRepository.delete(orderItem);
+            }
```
Comment (mbarzilai) [inline src/main/java/com/games/online/shop/service/dao/GameShopDao.java:94]: Again, what should happen if the order isn't present?

File: src/test/java/com/games/online/shop/service/dao/GameShopDaoTest.java [hunk 4]
```diff
@@ -88,6 +99,37 @@ public void testDeleteMechanicsCascading() {
...
+        OrderItemId orderItemId = new OrderItemId(order.getId(), orderItem.getGame().getId());
+        orderItem.setId(orderItemId);
+        orderItem.setOrder(order);
+
+        order.addOrderItem(orderItem);
+        orderItemRepository.save(orderItem);
+
+        gameShopDao.deleteOrderItem(order.getId(), orderItem);
+        Order orderFromDb = orderRepository.getOne(order.getId());
+        assertThat(orderFromDb.getOrderItems()).isEmpty();
```
Comment (mbarzilai) [inline src/test/java/com/games/online/shop/service/dao/GameShopDaoTest.java:134]: Have you confirmed in your manual testing that the game from the order item isn't deleted? Do we want to add that to this test somehow?

File: src/main/java/com/games/online/shop/service/dao/GameShopDao.java (hunk 2 above)
Comment (madisoncarr) [reply src/main/java/com/games/online/shop/service/dao/GameShopDao.java:79]: Good catch! Fixed 👍

**Review by mbarzilai — APPROVED** [review]

**Pull request merged**